File: tests/xcalablemp/global-view/util/C-Round/xmp_mpi.c
#include <stdio.h>
#include "mpi.h"
#include "xmp.h"

extern int sub_mpi(MPI_Comm *comm);

#pragma xmp nodes p(4)

int main(int argc, char **argv) {

  xmp_init_mpi(&argc, &argv);
  int rank, size;

  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
  MPI_Comm_size(MPI_COMM_WORLD, &size);

#pragma xmp task on p(2:3)
{
  MPI_Comm comm;
  comm = xmp_get_mpi_comm();
  sub_mpi(&comm);
}

  xmp_finalize_mpi();

  return 0;
}

File: tests/XMP/global-view/util/F/xmp_mpi.f90
program main
  integer rank, irank, isize, ierr
  integer xmp_get_mpi_comm
  integer comm

!$xmp nodes p(4)

  call xmp_init_mpi()

!$xmp task on p(2:3)
  comm = xmp_get_mpi_comm()
  call sub_mpi(comm)
!$xmp end task

  call xmp_finalize_mpi()

end program
